### PR TITLE
[Update model]: Fleet.vehicles add optional components property

### LINKED ISSUE #712: [Model Update]: fleet.vehicles: Insert components list
## Update Reason
During test of data model BMW added the request to add components entity into the model.
The new property and entity is set to optional to allow backward compatibility.

## MS1 Criteria

- [x] The model that should be updated exists
- [x] The referenced use case exists -> QAX CX use case quality
- [x] The potential updates are discussed with all stakeholders -> during PI planning
- [x] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [x] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by TorstenHirsch**
## Description
Inside QAX we checked the latest version of this model. In a bilateral data exchange with real data BMW and Bosch figured out that one optional property is missing components. This property is a list of components and should be used to store identifier for focus components. This helps use quality to overcome the situation that is currently unclear which serial part number of a focus component is installed in a vehicle/ is linked to a hashed VIN.

Closes #712 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.6.0)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider: BMW, Volkswagen
- [x] The new aspect (version) will be consumed by at least one data consumer: Bosch, SAP, ZF,
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by TorstenHirsch** [comment]
@agg3fe  Please review - Thank you

**Comment by TorstenHirsch** [comment]
modeling_team

**Comment by TorstenHirsch** [comment]
@agg3fe Could you check the model and confirm MS2? I have to check whether I can extract on Friday component enity to a shared model - but if I do not succeed in alignment I definitely need the change in the model with shared model approach

**Comment by TorstenHirsch** [comment]
modeling_team

File: io.catenax.fleet.vehicles/3.0.0/Vehicles.ttl [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+   samm-c:constraint :CountryCodeRegularExpression .
+
+:VehicleSteeringPos a samm-c:Enumeration ;
+   samm:preferredName "Vehicle steering position(NHTSA)"@en ;
+   samm:description "Vehicle steering position enumeration from NHTSA, see table [vPICList_lite].[dbo].[Steering]"@en ;
+   samm:see <https://vpic.nhtsa.dot.gov/api/> ;
+   samm:dataType xsd:string ;
+   samm-c:values ( "Left-Hand Drive" "Right-Hand Drive" ) .
+
+:OemShortNameTrait a samm-c:Trait ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/3.0.0/Vehicles.ttl:282]: preferredName and description missing

File: io.catenax.fleet.vehicles/3.0.0/Vehicles.ttl [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+   samm:dataType xsd:string ;
+   samm-c:values ( "All-Wheel Drive" "Front-Wheel Drive" "Rear-Wheel Drive" ) .
+
+:PowerTrainTypes a samm-c:Enumeration ;
+   samm:preferredName "Powertrain types "@en ;
+   samm:description "This enumeration comes from NHTSA (US authority) database. Table [vPICList_lite].[dbo].[ElectrificationLevel] extended by Gasoline and Diesel as entry."@en ;
+   samm:dataType xsd:string ;
+   samm-c:values ( "BEV (Battery Electric Vehicle)" "Diesel" "FCEV (Fuel Cell Electric Vehicle)" "Gasoline" "HEV (Hybrid Electric Vehicle)" "Mild HEV (Hybrid Electric Vehicle)" "PHEV (Plug-in Hybrid Electric Vehicle)" "Other" ) .
+
+:CountryCodeTrait a samm-c:Trait ;
```
Comment (agg3fe) [inline io.catenax.fleet.vehicles/3.0.0/Vehicles.ttl:271]: preferredName and description missing

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check few comments

**Review by agg3fe — APPROVED** [review]
LGTM

**Comment by TorstenHirsch** [comment]
Hi @agg3fe , when will the merge happen?

**Pull request merged**